src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (table.length * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+        if (currentNode == null) {
+            table[index] = new Node<>(key, value);
+        } else {
+            for (; currentNode != null; currentNode = currentNode.next) {
+                if (Objects.equals(key, currentNode.key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = new Node<>(key, value);
+                    break;
+                }
+            }
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> x = table[getIndex(key)]; x != null; x = x.next) {
+            if (Objects.equals(x.key, key)) {
+                return x.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    private int getIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        int oldCap = oldTable.length;
+        if (oldCap > 0) {
+            int newCap = oldCap << 1;
+            threshold = threshold << 1;
+            size = 0;
+            Node<K, V>[] newTable = (Node<K, V>[]) new Node[newCap];
+            table = newTable;
+            transfer(oldTable);
+        }
+    }
+
+    private void transfer(Node<K, V>[] oldTable) {
+        for (Node<K, V> node : oldTable) {
+            for (Node<K, V> first = node; first != null; first = first.next) {
+                put(first.key, first.value);
+            }
```

Comment:
We do not need `first `
```suggestion
            while (node != null) {
                put(node.key, node.value);
                node = node.next;
            }
```